app/src/ui/clone-repository/clone-github-repository.tsx

```diff
@@ -114,6 +117,12 @@ export class CloneGithubRepository extends React.Component<
   }
 
   public componentWillReceiveProps(nextProps: ICloneGithubRepositoryProps) {
+    if (nextProps.shouldClearFilter) {
```

Comment:
To ensure this is only applied once, when `shouldClearFilter` transitions from `false` to `true`, we should do something like this:

```tsx
if (this.props.shouldClearFilter === false && nextProps.shouldClearFilter) { 
```

Reply:
Hmm, maybe I'm misunderstanding, because I think we actually need this to be executed every time the tab is changed, otherwise, like you said, it will only be applied once, so from the second time and on that we switch tabs we'll have the original behavior of the bug, like this:

![clear-filter](https://user-images.githubusercontent.com/7514993/35860711-6a615a38-0b1b-11e8-9bff-93dcc57f34b5.gif)

Is this what's desired? :/

Reply:
I think that what you are aiming for is to persist the filter text of each tab among switches? If that's the case check this comment https://github.com/desktop/desktop/pull/3787#issuecomment-357472645 and let me know!

Reply:
@reyronald 

> Hmm, maybe I'm misunderstanding, because I think we actually need this to be executed every time the tab is changed, otherwise, like you said, it will only be applied once, so from the second time and on that we switch tabs we'll have the original behavior of the bug ...

The subsequent switching between tabs (which is driven from `app.tsx` and flows through props changes on `clone-repository.tsx`) should trigger this code:

```tsx
  public componentWillReceiveProps(nextProps: ICloneRepositoryProps) {
    this.setState({
      shouldClearFilter: this.props.selectedTab !== nextProps.selectedTab,
    })
  }
```

Which should then trigger `clone-repository.tsx` to re-render with `shouldClearFilter=true`.

I wanted to avoid unnecessarily clearing the filter text or unnecessary rendering, but I think it's a minor thing. I couldn't break it either with this change or my proposed refinement.

Reply:
@shiftkey Yeah, it does trigger that code which then triggers a re-render of `clone-github-repository.tsx` with a `shouldClearFilter={true}` just like you said, but in those subsequent calls `this.props.shouldClearFilter` would be `true`, so that condition (the `!this.props.shouldClearFilter && nextProps.shouldClearFilter`) would always evaluate to false from then on. 

The GIF I uploaded in my previous comment is the behavior with the suggested change, did you test it yourself and didn't see that happening?  Or perhaps there's something fishy with my dev env 😅

Reply:
> The GIF I uploaded in my previous comment is the behavior with the suggested change, did you test it yourself and didn't see that happening? 

My GHE instance needs to do browser signin - and it rejects the developer OAuth app - so I haven't been testing the exact steps you were. Let me test it in non-dev mode :thumbsup:

Reply:
And yep, I now see the same behaviour you do with my proposed change.

Nevermind, let's stick with yours :metal:

Reply:
Great! ^^, glad we didn't let that slip 💪  !

Just pushed the changes, do you want me to squash the commits? @shiftkey

Reply:
@reyronald don't worry about it. I'll take it from here.